src/Config/Converters/EntityRestOptionsConverter.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.DataApiBuilder.Config.ObjectModel;
+
+namespace Azure.DataApiBuilder.Config.Converters;
+
+internal class EntityRestOptionsConverter : JsonConverter<EntityRestOptions>
+{
+    /// <inheritdoc/>
+    public override EntityRestOptions? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        if (reader.TokenType == JsonTokenType.StartObject)
+        {
+            EntityRestOptions restOptions = new(Methods: Array.Empty<SupportedHttpVerb>(), Path: null, Enabled: true);
+            while (reader.Read())
+            {
+                if (reader.TokenType == JsonTokenType.EndObject)
+                {
+                    break;
+                }
+
+                string? propertyName = reader.GetString();
+
+                switch (propertyName)
+                {
+                    case "path":
+                    {
+                        reader.Read();
+
+                        if (reader.TokenType == JsonTokenType.String)
+                        {
+                            restOptions = restOptions with { Path = reader.DeserializeString() };
+                            break;
+                        }
+
+                        if (reader.TokenType == JsonTokenType.True || reader.TokenType == JsonTokenType.False)
```

Comment:
I dont think we intended to support path = true/false as per the JSON schema, this may be a bug in existing implementation.

Reply:
Yes, looking at the JSON schema you're correct. Deleted that condition and now throwing an error if the value is invalid.

Reply:
Reopening as I encountered that my local config, and config files present in the test project have `"path": true` used, which results in an engine startup failure.

Schema file is accurate to only support 'string' as you two mention: https://github.com/Azure/data-api-builder/blob/90229412a221cb0caaefd450fca57debd4bc9bf9/schemas/dab.draft.schema.json#L286C15-L288

I'll submit a PR to fix the files in the test project (so we don't increase file count here). 

I confirming whether the CLI tool properly adheres to the schema requirement so we don't have to tag this as a breaking change.

Reply:
Created #1547 to track this work. Shall we resolve the comment are refactor after?

Reply:
Can close this thread. I added a PR which addresses the improper config in the test file. #1549